### PR TITLE
Test that the rust doc matches README.md

### LINKED ISSUE #5: Figure out if there's a way to import the README into `src/lib.rs`
Per @npmccallum: https://github.com/enarx/iocuddle/pull/4#discussion_r463150742

### PR DESCRIPTION
**Pull request by mbestavros**
Resolves #5

### CODE REVIEW

**Comment by mbestavros** [comment]
It would indeed.

The doc comment on the top of `src/lib.rs` should be a verbatim copy of the README, basically.

**Comment by lkatalin** [comment]
@mbestavros @npmccallum Maybe this is just because I'm naive to some Rust crate conventions, but it seems like it wouldn't always be desirable to have these be exact copies of each other? What about in the case of something like [sallyport](https://github.com/enarx/enarx/tree/master/sallyport) where there's a lot of extra, very useful info in the `readme` that wouldn't fit that well into `src/lib.rs` because it's more of an overview? Or would we just want to copy all that info over to `src/lib.rs` anyway?

File: src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,9 @@
 // SPDX-License-Identifier: Apache-2.0
 
+//! ![test](https://github.com/enarx/iocuddle/workflows/test/badge.svg)
+//! ![lint](https://github.com/enarx/iocuddle/workflows/lint/badge.svg)
+//! ![rust v1.36+](https://img.shields.io/badge/rustc-v1.36%2B-blue.svg)
```
Comment (npmccallum) [inline src/lib.rs:5]: I think we want to follow the badges workflow outlined here: https://github.com/livioribeiro/cargo-readme

**Comment by haraldh** [comment]
> @mbestavros @npmccallum Maybe this is just because I'm naive to some Rust crate conventions, but it seems like it wouldn't always be desirable to have these be exact copies of each other? What about in the case of something like [sallyport](https://github.com/enarx/enarx/tree/master/sallyport) where there's a lot of extra, very useful info in the `readme` that wouldn't fit that well into `src/lib.rs` because it's more of an overview? Or would we just want to copy all that info over to `src/lib.rs` anyway?

Oh, I really like to read the crate docs with all the info needed, without going to the repo and read another additional document.

**Comment by lkatalin** [comment]
@haraldh So in a normal case would the info from the Sallyport `readme` be transferred to `src/lib`? I just don't want to lose information.

**Comment by haraldh** [comment]
> @haraldh So in a normal case would the info from the Sallyport `readme` be transferred to `src/lib`? I just don't want to lose information.

Yes, I would do that.

File: src/lib.rs (hunk 1 above)
Comment (haraldh) [reply src/lib.rs:5]: yes please

Comment (haraldh) [reply src/lib.rs:5]: Hmm, for the `[rust v1.36+]` badge, we would have to use a `README.tpl` template, because it's not a Cargo.toml standard.

Comment (haraldh) [reply src/lib.rs:5]: you might want to have https://github.com/enarx/iocuddle/pull/7 first

**Comment by haraldh** [comment]
> Badge workflow should be changed.

https://github.com/enarx/iocuddle/pull/7
for some initial badges in Cargo.toml

**Comment by npmccallum** [comment]
@mbestavros #7 is merged now. So let's rebase on that.

**Pull request merged**